clients/python/build/prepare-release.sh

```diff
@@ -0,0 +1,170 @@
+    exit 1
+  fi
+}
+
+# Function: update_init_version
+# Description:
+#   Specifically updates the __version__ variable in a Python __init__.py file.
+# Parameters:
+#   $1 - Path to the __init__.py file.
+#   $2 - The new version string.
+update_init_version() {
+  local file_path="$1"
+  local new_version="$2"
+  local temp_file
+
+  if [[ -f "$file_path" ]]; then
+    temp_file=$(mktemp)
+    while IFS= read -r line; do
+      # If the line defines __version__, replace it with the new version.
+      if [[ $line =~ ^__version__[[:space:]]*=[[:space:]]*\"[^\"]+\" ]]; then
+        echo "__version__ = \"$new_version\"" >> "$temp_file"
+      else
+        echo "$line" >> "$temp_file"
+      fi
+    done < "$file_path"
+    mv "$temp_file" "$file_path"
+    echo "Updated __version__ in $file_path"
+  else
+    echo -e "${RED}${BOLD}Error:${NC} File not found: $file_path" >&2
+    exit 1
+  fi
+}
+
+# Function: validate_version_in_file
+# Description:
+#   Validates that the version number in a specified file matches the expected version.
+# Parameters:
+#   $1 - Path to the file to validate.
+#   $2 - Regex pattern to extract the version.
+#   $3 - The expected version string.
+validate_version_in_file() {
+  local file_path="$1"
+  local pattern="$2"
+  local expected_version="$3"
+  local actual_version
+
+  if [[ -f "$file_path" ]]; then
+    actual_version=$(grep -E "$pattern" "$file_path" | head -n1 | awk -F '"' '{print $2}')
+    if [[ "$actual_version" == "$expected_version" ]]; then
+      echo -e "Validation passed for $file_path: version is ${BOLD}${GREEN}$expected_version${NC}"
+    else
+      echo -e "${RED}${BOLD}Error:${NC} Validation failed for $file_path. Expected version: $expected_version, Found: $actual_version" >&2
+      exit 1
+    fi
+  else
+    echo -e "${RED}${BOLD}Error:${NC} File not found during validation: $file_path" >&2
+    exit 1
+  fi
+}
+
+# -------------------- Main Script Logic --------------------
+
+# Check if exactly one argument (release version) is provided.
+if [[ $# -ne 1 ]]; then
+  echo -e "${RED}${BOLD}Error:${NC} Release version not provided."
+  usage
+fi
+
+# Assign the first argument to RELEASE_VERSION variable.
+RELEASE_VERSION="$1"
+
+# Version validation regex to accept formats like:
+# 1.2.3, 1.2.3-beta, 1.2.3beta, 1.2.3.beta, 1.2.3rc1, 1.2.3.dev0, etc.
+if [[ ! "$RELEASE_VERSION" =~ ^[0-9]+\.[0-9]+\.[0-9]+([.-]?[A-Za-z0-9]+)?$ ]]; then
+  echo -e "${RED}${BOLD}Error:${NC} ${RED}Invalid release version format.${NC}"
+  usage
+fi
+
+# Define the root directory of the repository.
+REPO_ROOT="$(pwd)"
+
+# Define paths to the files that need to be updated with the new version.
+PYPROJECT_PATH="$REPO_ROOT/clients/python/target/pyproject.toml"
+SETUP_PY_PATH="$REPO_ROOT/clients/python/target/setup.py"
+INIT_PY_PATH="$REPO_ROOT/clients/python/target/src/unitycatalog/client/__init__.py"
+
+# -------------------- Updating Version Files --------------------
+
+# Update version in pyproject.toml
```

Comment:
Why not use whatever build tool is in use? `hatch` and `uv` and `poetry` all support it.

E.g., https://hatch.pypa.io/latest/version/